src/Microsoft.Health.Fhir.Shared.Api/Controllers/OperationDefinitionController.cs

```diff
@@ -0,0 +1,137 @@
+        {
+            EnsureArg.IsNotNull(mediator, nameof(mediator));
+            EnsureArg.IsNotNull(operationsConfig?.Value, nameof(operationsConfig));
+            EnsureArg.IsNotNull(featureConfig?.Value, nameof(featureConfig));
+
+            _mediator = mediator;
+            _operationConfiguration = operationsConfig.Value;
+            _featureConfiguration = featureConfig.Value;
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.ReindexOperationDefinition, Name = RouteNames.ReindexOperationDefintion)]
+        [AllowAnonymous]
+        public async Task<IActionResult> ReindexOperationDefintion()
+        {
+            return await GetOperationDefinitionAsync(OperationsConstants.Reindex);
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.ResourceReindexOperationDefinition, Name = RouteNames.ResourceReindexOperationDefinition)]
+        [AllowAnonymous]
+        public async Task<IActionResult> ResourceReindexOperationDefinition()
+        {
+            return await GetOperationDefinitionAsync(OperationsConstants.ResourceReindex);
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.ExportOperationDefinition, Name = RouteNames.ExportOperationDefinition)]
+        [AllowAnonymous]
+        public async Task<IActionResult> ExportOperationDefinition()
+        {
+            return await GetOperationDefinitionAsync(OperationsConstants.Export);
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.PatientExportOperationDefinition, Name = RouteNames.PatientExportOperationDefinition)]
+        [AllowAnonymous]
+        public async Task<IActionResult> PatientExportOperationGetDefinition()
+        {
+            return await GetOperationDefinitionAsync(OperationsConstants.PatientExport);
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.GroupExportOperationDefinition, Name = RouteNames.GroupExportOperationDefinition)]
+        [AllowAnonymous]
+        public async Task<IActionResult> GroupExportOperationDefinition()
+        {
+            return await GetOperationDefinitionAsync(OperationsConstants.GroupExport);
+        }
+
+        [HttpGet]
+        [Route(KnownRoutes.AnonymizedExportOperationDefinition, Name = RouteNames.AnonymizedExportOperationDefinition)]
+        [AllowAnonymous]
+        public async Task<IActionResult> AnonymizedExportOperationDefinition()
+        {
+            return await GetOperationDefinitionAsync(OperationsConstants.AnonymizedExport);
+        }
+
+        private async Task<IActionResult> GetOperationDefinitionAsync(string operationName)
+        {
+            CheckIfOperationIsEnabledAndRespond(operationName);
+
+            OperationDefinitionResponse response = await _mediator.GetOperationDefinitionAsync(operationName, HttpContext.RequestAborted);
+
+            return FhirResult.Create(response.OperationDefinition, HttpStatusCode.OK);
+        }
+
+        private void CheckIfOperationIsEnabledAndRespond(string operationName)
```

Comment:
Does it matter if the definitions are always available or not?

Reply:
I don't think there is anything in the spec that says we shouldn't have these always available. I just felt it might be consistent to have it available only when the relevant operations are enabled. I am open to having them available all the time.